test/trace/basic_server.js

```diff
@@ -163,3 +163,17 @@ test('basic tracing test', function (assert) {
         }, 10);
     });
 });
+
+test('modifying traceSample', function (assert){
+    var channel = new TChannel();
+    var subservice = channel.makeSubChannel({
+        serviceName: 'subservice'
+    });
+    assert.equal(channel.traceSample, 0.01);
+    assert.equal(subservice.traceSample, 0.01);
+
+    subservice.setTraceSample(0.99);
+    assert.equal(channel.traceSample, 0.99);
+    assert.equal(subservice.traceSample, 0.99);
```

Comment:
The proposed semantics appear to be that all channels share a traceSample property with their top channel. What synchronizes the traceSample across peer subchannels? Should the trace sample property only exist on the top channel and be null for other channels? Should the trace sample property of subchannels be a getter that reaches into the top channel?